frontend/src/components/layout/PrivateRoute.js

```diff
@@ -0,0 +1,27 @@
+import React, { useContext } from 'react'
+import { Route, Redirect } from 'react-router-dom'
+import AuthContext from '../../context/AuthContext';
+
+// A wrapper for <Route> that redirects to the login
+// screen if you're not yet authenticated.
+export default function PrivateRoute({ children, ...rest }) {
+    const { loggedIn } = useContext(AuthContext)
+  
+    return (
+        <Route
+        {...rest}
+        render={({ location }) =>
+            loggedIn? (
```

Comment:
Nitpick: add a space before the question mark so it's easier to spot that it's an ternary expression.